openmc/tallies.py

```diff
@@ -1410,6 +1432,765 @@ def export_results(self, filename='tally-results', directory='.',
             # Pickle the Tally results to a file
             pickle.dump(tally_results, open(filename, 'wb'))
 
+    def _outer_product(self, other, new_tally, binary_op):
+        """Combines filters, scores and nuclides with another tally.
+
+        This is a helper method for the tally arithmetic methods. The filters,
+        scores and nuclides from both tallies are enumerated into all possible
+        combinations and expressed as CrossFilter, CrossScore and
+        CrossNuclide objects in the new derived tally.
+
+        Parameters
+        ----------
+        other : Tally
+            The tally on the right hand side of the outer product
+        new_tally: Tally
+            The new tally to represent the outer product
+        op : str
```

Comment:
The numpydoc syntax for a limited set of choices would be:
`binary_op : {'+', '-', '*', '/', '^'}`